src/Manager.php

```diff
@@ -231,18 +233,7 @@ public function custom($name, array $config = [])
      */
     public function loadFilter($name, $filter, array $options = [])
     {
-        if (empty($options['className'])) {
-            $class = Inflector::classify($filter);
-        } else {
-            $class = $options['className'];
-            unset($options['className']);
-        }
-        $className = App::className($class, 'Model\Filter');
-        if (!$className) {
-            throw new InvalidArgumentException(sprintf('Search filter "%s" was not found.', $class));
-        }
-
-        return new $className($name, $this, $options);
+        return (new FilterLocator($this))->locate($name, $filter, $options);
```

Comment:
Why build a new locator instance for each filter?  A single locator instance should suffice.

Unless the locator class is going to have more methods having a class with single method seems a bit of over engineering. In PR #221 you mentioned adding a `FilterCollection` (I like that idea). So couldn't that class handle locating/building instance too? Similar to what `WidgetLocator` does in core.

Reply:
Also, using a hardcoded and non-factorized `(new FilterLocator($this))` seems like an anti-pattern to me.
This should at least then be a protected method `locator()` etc so you can exchange this if needed for your own locator when extending the manager etc.

Reply:
@ADmad I guess you want to drag that for BC reasons along anyway, so can't we just stick with the locator and use it in the manager and in the collection when it's done?